build-llvm34.md

```diff
@@ -68,79 +61,126 @@ If you want to build KLEE with LLVM 2.9, [click here]({{site.baseurl}}/build-llv
 
    ```bash
    $ git clone https://github.com/klee/klee.git
    ```
 
-7. **Configure KLEE:** From the KLEE source directory, run:  
+9. **Configure KLEE:**
+
+   KLEE must be built "out of source" so first make a binary build directory. You can create this where ever you like.
 
    ```bash
-   $ ./configure --with-stp=/full/path/to/stp/build --with-uclibc=/full/path/to/klee-uclibc --enable-posix-runtime
+   $ mkdir klee_build_dir
    ```
 
-   **NOTE:** If LLVM is not found or you have multiple LLVM versions installed, you can add `--with-llvmsrc=/usr/lib/llvm-3.4/build --with-llvmobj=/usr/lib/llvm-3.4/build --with-llvmcc=/usr/bin/clang-3.4 --with-llvmcxx=/usr/bin/clang++-3.4`.  
-If you skipped step 4, simply remove the `--with-uclibc` and `--enable-posix-runtime` options.<br/><br/>  
+   Now `cd` into the build directory and run CMake to configure KLEE where `<KLEE_SRC_DIRECTORY>` is the path
+   to the KLEE git repository you cloned in step 7.
 
-8. **Build KLEE:**  
+   ```bash
+   $ cd klee_build_dir
+   $ cmake <CMAKE_OPTIONS> <KLEE_SRC_DIRECTORY>
+   ```
+
+   `<CMAKE_OPTIONS>` are any of the options that were mentioned as required for previous steps.
+   For example if KLEE was being built with STP, the POSIX runtime, klee-uclibc and testing then the
+   command line would look something like this
 
    ```bash
-   $ make  
+   cmake -DENABLE_SOLVER_STP=ON \
+     -DENABLE_POSIX_RUNTIME=ON \
+     -DENABLE_KLEE_UCLIBC \
+     -DKLEE_UCLIC_PATH=<KLEE_UCLIC_SOURCE_DIR> \
+     -DGTEST_SRC_DIR=<GTEST_SOURCE_DIR> \
+     <KLEE_SRC_DIRECTORY>
    ```
-   <!-- make DISABLE_ASSERTIONS=0 ENABLE_OPTIMIZED=1 ENABLE_SHARED=0 -j2-->
 
-   **NOTE:** You can add `/full/path/to/klee/build/Release/bin` to your path.<br/><br/>
+   There are many KLEE configuration options see the [README-CMake.md](https://github.com/klee/klee/blob/master/README-CMake.md) for the documentation on them.
 
+   **NOTE:** If LLVM is not found or you need a particular version to be used you can pass `-DLLVM_CONFIG_BINARY=<LLVM_CONFIG_BINARY>` to CMake where `<LLVM_CONFIG_BINARY>` is the absolute path to the
+   relevant `llvm-config` binary. Similary KLEE needs a C and C++ compiler that can create LLVM bitcode that is compatible with the version of LLVM KLEE is using. If these are not detected automatically `-DLLVMCC=<PATH_TO_CLANG>` and `-DLLVMCXX=<PATH_TO_CLANG++>` can be passed to explicitly set these compilers where `<PATH_TO_CLANG>` is the absolute path to `clang` and `<PATH_TO_CLANG++>` is the absolute path to `clang++`.
 
-9. **Run the main regression test suite to verify your build:**
+
+9. **Build KLEE:**  
+   From the ``klee_build_dir`` directory created in the previous step run.
+
+   ```bash
+   $ make  
+   ```
+
+9. **(Optional) Run the main regression test suite**
+   If you installed `lit` and KLEE was configured with integration tests enabled
+   then you can run them like this.
    
    ```bash
-   $ make test
+   $ make integrationtests
```

Comment:
The CMake build system uses a different target name. This was a deliberate choice because `test` is a very vague target name. The old build system had the unit test target called `unittests` so I preserved that name for the unit tests and used a similar name for the integration tests.